### PR TITLE
fix: empty tuples are correctly parsed

### LINKED ISSUE #589: Empty tuple default value not supported
## 🐛 Bug report

Empty tuples are not supported by the parser.

### To reproduce

```
from dataclasses import dataclass

from jsonargparse import ArgumentParser


@dataclass
class TestCfg:
    a: tuple[int, ...] = ()


if __name__ == "__main__":
    parser = ArgumentParser(description="CLI for training models", exit_on_error=False)
    parser.add_argument("test", type=TestCfg)
    args = parser.parse_args()
    print(args)
```
Running the following program without passing any arguments or by passing an empty list/tuple `"[]"` results in the following error:
```
argparse.ArgumentError: Validation failed: Parser key "test.a":
  Expected a non-empty tuple. Got value: []
  ```

### Expected behavior

To be able to parse empty tuples without errors.

### Environment

- jsonargparse version: 4.33.1
- Python version: 3.10
- How jsonargparse was installed: pip
- OS: MacOS

### PR DESCRIPTION
**Pull request by MiguelMonteiro**
## What does this PR do?

Empty tuples are now successfully parsed.

Fixes #589

## Before submitting

- [x] Did you read the [contributing guideline](https://github.com/omni-us/jsonargparse/blob/main/CONTRIBUTING.rst)?
- [ ] Did you update **the documentation**? (readme and public docstrings)
- [x] Did you write **unit tests** such that there is 100% coverage on related code? (required for bug fixes and new features)
- [x] Did you verify that new and existing **tests pass locally**?
- [x] Did you make sure that all changes preserve **backward compatibility**?
- [x] Did you update **the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

### CODE REVIEW

File: jsonargparse/_typehints.py [hunk 1]
```diff
@@ -848,7 +848,7 @@ def adapt_typehints(
                 subtypehint = subtypehints[0 if is_ellipsis or not is_tuple else n]
                 val[n] = adapt_typehints(v, subtypehint, **adapt_kwargs)
             if is_tuple and len(val) == 0:
-                raise_unexpected_value("Expected a non-empty tuple", val)
```
Comment (mauvilsa) [inline jsonargparse/_typehints.py:851]: Don't remove the raise for empty tuple. This is correct and necessary. Only tuples with `...` should be allowed to be empty. But other tuples not. The code here should be:
```python
if is_tuple and len(val) == 0 and not is_ellipsis:
    raise_unexpected_value("Expected a non-empty tuple", val)
```

**Review by mauvilsa — CHANGES_REQUESTED** [review]

File: jsonargparse/_typehints.py (hunk 1 above)
Comment (mauvilsa) [reply jsonargparse/_typehints.py:851]: Actually no. These two lines should be removed. Number of elements in the tuple are already handled by
```python
if is_tuple and not is_ellipsis and len(val) != len(subtypehints):
    raise_unexpected_value(f"Expected a tuple with {len(subtypehints)} elements", val)
```
And converting the empty list to a tuple would already be handled by
```python
if not serialize:
    val = tuple(val) if typehint_origin in {Tuple, tuple} else set(val)
```

**Review by mauvilsa — CHANGES_REQUESTED** [review]
Also please add a changelog entry in a new section as explained in the contributions guide.

**Comment by MiguelMonteiro** [comment]
Thank you for the feedback, I have made the requested changes, let me know if it's correct.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -12,6 +12,15 @@ The semantic versioning only considers the public API as described in
 paths are considered internals and can change in minor and patch releases.
 
 
+v4.33.3 (2024-10-09)
```
Comment (mauvilsa) [inline CHANGELOG.rst:15]:
```suggestion
v4.33.3 (2024-10-??)
```

**Review by mauvilsa — CHANGES_REQUESTED** [review]

**Pull request merged**